tools/lib/databaseToTileset.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+var Cesium = require('cesium');
+var fsExtra = require('fs-extra');
+var path = require('path');
+var Promise = require('bluebird');
+var sqlite3 = require('sqlite3');
+var zlib = require('zlib');
+var isGzipped = require('../lib/isGzipped');
+
+var fsExtraOutputFile = Promise.promisify(fsExtra.outputFile);
+
+var defaultValue = Cesium.defaultValue;
+var defined = Cesium.defined;
+var DeveloperError = Cesium.DeveloperError;
+
+module.exports = tilesetToDatabase;
+
+function tilesetToDatabase(inputFile, outputDirectory) {
+    if (!defined(inputFile)) {
+        throw new DeveloperError('inputFile is required.');
+    }
+
+    outputDirectory = defaultValue(outputDirectory, path.join(path.dirname(inputFile), path.basename(inputFile, path.extname(inputFile))));
+
+    // Open the database.
+    var db = new sqlite3.Database(inputFile, sqlite3.OPEN_READWRITE);
+    var dbGet = Promise.promisify(db.get, {context : db});
+
+    // Get number of rows
+    return dbGet('SELECT Count(*) AS total FROM media')
+        .then(function(row) {
+            return row.total;
+        })
+        .then(function(total) {
+            var promises = new Array(total);
+            for (var i = 0; i < total; ++i) {
+                promises[i] = readAndWriteFile(outputDirectory, dbGet, i);
+            }
+            return Promise.all(promises);
+        })
+        .finally(function() {
+            db.close();
+        });
+}
+
+function readAndWriteFile(outputDirectory, dbGet, index) {
+    return dbGet('SELECT * FROM media LIMIT 1 OFFSET ?', index)
```

Comment:
Running a new query for each row isn't efficient. We should do a select in chunks. Maybe like run the query to grab in 10,000 row chunks. You want to be careful with the gzipping and file writing as to not thrash disk IO.

I've done something similar like this:
```
var offset = 0;
var limit = 10000; // Or some reasonable limit
var asyncLimit = ??; // Number of cores or whatever
var processChunk = function() {
    return dbAll('SELECT * FROM media LIMIT ? OFFSET ?', limit, offset).then(function(rows) {
        if (rows.length === 0) {
            // No more rows left
            return Promise.resolve();
        }
        
        return Promise.map(rows, function(row) {
            // TODO: Handle gzipping/file writing
            ++offset;
        }, {concurrency: asyncLimit})
            .then(function () {
                return processChunk();
            });
    });
};

return processChunk();
```